tapgarden/caretaker.go

```diff
@@ -1188,3 +1189,113 @@ func GenHeaderVerifier(ctx context.Context,
 		return err
 	}
 }
+
+// GenGroupVeifier generates a group key verification callback function given
+// a DB handle.
+func GenGroupVerifier(ctx context.Context,
+	mintingStore MintingStore) func(*btcec.PublicKey) error {
+
+	// Cache known group keys that were previously fetched.
+	assetGroups := fn.NewSet[asset.SerializedKey]()
+
+	return func(groupKey *btcec.PublicKey) error {
+		if groupKey == nil {
+			return fmt.Errorf("cannot verify empty group key")
+		}
+
+		assetGroupKey := asset.ToSerialized(groupKey)
```

Comment:
Since we don't know if this is accessed by goroutines in parallel, do we need to add a mutex to guard the cache as well?

Reply:
Good point, in the global config I made a separate once for each subsystem, but I forgot that some of these paralellize internally.